src/TerraformEnvironmentStage.groovy

```diff
@@ -14,6 +12,7 @@ class TerraformEnvironmentStage implements Stage {
     public static final String PLAN = 'plan'
     public static final String CONFIRM = 'confirm'
     public static final String APPLY = 'apply'
+    public static final String DESTROY = 'destroy'
```

Comment:
This new state of "DESTROY".  How will this behavior with the existing plugins that are part of  the library (or worse, plugins that are in completely external libraries?)

eg: 

* https://github.com/manheim/terraform-pipeline/blob/74de97b89fdedf087fde028686b1e6530a30aed2/src/ParameterStoreExecPlugin.groovy#L14-L20
* https://github.com/manheim/terraform-pipeline/blob/74de97b89fdedf087fde028686b1e6530a30aed2/src/AnsiColorPlugin.groovy#L11-L14
* https://github.com/manheim/terraform-pipeline/blob/817be51f85223d535120fb7d17e35698f34d8f17/src/ParameterStoreBuildWrapperPlugin.groovy#L17-L29
* https://github.com/manheim/terraform-pipeline/blob/00e3eed6cfcf8e4ad40a68a323f116202cda90da/src/CrqPlugin.groovy#L11-L16

Reply:
@kmanning I am making some changes now. I will have to add `stage.decorate(DESTROY, someMethod(options))` to the above plugins. This will add functionality if DESTROY is enabled. Otherwise this will not change the existing functionality.